go/pkg/framework/server.go

```diff
@@ -23,28 +23,36 @@ import (
 	"github.com/mlrun/mlrun/proto/build/health"
 
 	"github.com/nuclio/errors"
-	"github.com/nuclio/loggerus"
+	"github.com/nuclio/logger"
 	"google.golang.org/grpc"
 )
 
 const HealthWatchInterval = 5 * time.Second
 
 type MlrunGRPCServer interface {
-	getLogger() *loggerus.Loggerus
-	setServer(*grpc.Server)
-	getServerOpts() []grpc.ServerOption
 	RegisterRoutes(context context.Context)
 	OnBeforeStart(context context.Context) error
+	getLogger() logger.Logger
+	setServer(*grpc.Server)
+	getServerOpts() []grpc.ServerOption
 }
 
 type AbstractMlrunGRPCServer struct {
-	Logger         *loggerus.Loggerus
+	Logger         logger.Logger
 	Server         *grpc.Server
 	servingStatus  health.HealthCheckResponse_ServingStatus
 	grpcServerOpts []grpc.ServerOption
 }
 
-func (s *AbstractMlrunGRPCServer) getLogger() *loggerus.Loggerus {
+func NewAbstractMlrunGRPCServer(logger logger.Logger, grpcServerOpts []grpc.ServerOption) (*AbstractMlrunGRPCServer, error) {
+	return &AbstractMlrunGRPCServer{
+		Logger:         logger,
```

Comment:
What do you mean "namespace the logger"?

Reply:
when you use `GetChild` you create a sublogger. eventually, you get an hierarchy of loggers. this will help you understand which component logs "main.http.client" vs "main" when you log from main/http/client package.